stacker/providers/aws/default.py

```diff
@@ -94,6 +94,9 @@ def __init__(self, region, **kwargs):
         # Necessary to deal w/ multiprocessing issues w/ sharing ssl conns
         # see: https://github.com/remind101/stacker/issues/196
         self._pid = os.getpid()
+        # If region is not set explicitly, fetch it from the boto session
+        if not self.region:
```

Comment:
You make a good point about not allowing changing of the region - but I think we should be careful of squashing any potential settings people have/rely on.  It seems unlikely that people have a `region` variable, and that it'd be different than what they'd want when using `--region`, etc, but it might.

How about we throw a helpful exception if someone has `region` set manually?

Also, can you add some documentation that `region` is a reserved environment variable, and how it's set?